pkg/asset/machines/aws/instance_types.go

```diff
@@ -47,8 +47,14 @@ func PreferredInstanceType(ctx context.Context, meta *awsconfig.Metadata, types
 	}
 	for _, t := range types {
 		if reqZones.Difference(sets.NewString(found[t]...)).Len() == 0 {
-			return t, nil
+			return t, zones, nil
 		}
 	}
-	return types[0], errors.New("no instance type found for the zone constraint")
+	// HACK: in the ARM case, the m6g and m6gd instances are not available in all the zones. This might be OK as is, but the
+	// manifests generated list all the zones as available which is incorrect. Instead return only the available zones
+	// Delete the zones in which the instance is not available and return the rest
+	for k := range reqZones.Difference(sets.NewString(found[types[0]]...)) {
+		reqZones.Delete(k)
```

Comment:
This should only be done if the user did not specify which zones to use. If the user did specify the zones to use, either explicitly or implicitly via provided subnets, then the installation should fail if any of those zones cannot be used.

Reply:
https://github.com/openshift/installer/blob/296b0b10da2a5a9a5afca6c10b03ee73ce98578c/pkg/asset/installconfig/aws/validation.go#L178-L200

Reply:
ok..changed the logic to return a filtered list of zones only if the option has not been specified through subnets/zones in the install config.